File: admin/src/api/real-name.ts
import { http, type ApiEnvelope } from '../utils/request'

import type { PaginatedData } from './admin-user'

export interface RealNameUserSummary {
  id: number
  username: string
  email: string
  display_name: string | null
  status: string
}

export interface RealNameApplicationItem {
  id: number
  application_no: string
  user: RealNameUserSummary
  real_name: string
  id_type: string
  id_number_masked: string
  verification_provider: string | null
  provider_application_id: string | null
  provider_status: string | null
  provider_result_code: string | null
  provider_result_message: string | null
  provider_trace_id: string | null
  status: string
  submit_attempt: number
  failure_reason: string | null
  provider_started_at: string | null
  provider_finished_at: string | null
  created_at: string
  updated_at: string
}

export interface RealNameListQuery {
  page?: number
  per_page?: number
  keyword?: string
  status?: string
  id_type?: string
  provider?: string
  provider_status?: string
  date_from?: string
  date_to?: string
}

export interface RealNameReviewRequest {
  status: 'approved' | 'rejected'
  reason?: string
}

export async function getRealNameApplications(query?: RealNameListQuery) {
  const response = await http.get<ApiEnvelope<PaginatedData<RealNameApplicationItem>>>('/real-name-applications', { params: query })
  return response.data.data
}

export async function getRealNameApplication(id: number) {
  const response = await http.get<ApiEnvelope<RealNameApplicationItem>>(`/real-name-applications/${id}`)
  return response.data.data
}

export async function syncRealNameApplication(id: number) {
  const response = await http.post<ApiEnvelope<RealNameApplicationItem>>(`/real-name-applications/${id}/sync`)
  return response.data.data
}

export async function reviewRealNameApplication(id: number, payload: RealNameReviewRequest) {
  const response = await http.post<ApiEnvelope<RealNameApplicationItem>>(`/real-name-applications/${id}/review`, payload)
  return response.data.data
}

File: server/internal/usecase/admin/dto/real_name.go
package dto

import "time"

type RealNameApplicationListQuery struct {
	Page           int    `form:"page" validate:"omitempty,min=1"`
	PerPage        int    `form:"per_page" validate:"omitempty,min=1,max=100"`
	Keyword        string `form:"keyword" validate:"omitempty,max=96"`
	Status         string `form:"status" validate:"omitempty,oneof=pending approved rejected"`
	IDType         string `form:"id_type" validate:"omitempty,oneof=id_card"`
	Provider       string `form:"provider" validate:"omitempty,oneof=alipay wechat manual"`
	ProviderStatus string `form:"provider_status" validate:"omitempty,max=64"`
	DateFrom       string `form:"date_from" validate:"omitempty,max=32"`
	DateTo         string `form:"date_to" validate:"omitempty,max=32"`
}

type RealNameReviewRequest struct {
	Status string `json:"status" validate:"required,oneof=approved rejected"`
	Reason string `json:"reason" validate:"omitempty,max=500"`
}

type RealNameUserSummary struct {
	ID          uint64  `json:"id"`
	Username    string  `json:"username"`
	Email       string  `json:"email"`
	DisplayName *string `json:"display_name"`
	Status      string  `json:"status"`
}

type RealNameApplicationItem struct {
	ID                    uint64              `json:"id"`
	ApplicationNo         string              `json:"application_no"`
	User                  RealNameUserSummary `json:"user"`
	RealName              string              `json:"real_name"`
	IDType                string              `json:"id_type"`
	IDNumberMasked        string              `json:"id_number_masked"`
	VerificationProvider  *string             `json:"verification_provider"`
	ProviderApplicationID *string             `json:"provider_application_id"`
	ProviderStatus        *string             `json:"provider_status"`
	ProviderResultCode    *string             `json:"provider_result_code"`
	ProviderResultMessage *string             `json:"provider_result_message"`
	ProviderTraceID       *string             `json:"provider_trace_id"`
	Status                string              `json:"status"`
	SubmitAttempt         uint                `json:"submit_attempt"`
	FailureReason         *string             `json:"failure_reason"`
	ProviderStartedAt     *time.Time          `json:"provider_started_at"`
	ProviderFinishedAt    *time.Time          `json:"provider_finished_at"`
	CreatedAt             time.Time           `json:"created_at"`
	UpdatedAt             time.Time           `json:"updated_at"`
}
